### PR TITLE
Medical Adjudicator Session View and Report View for Database Integration Branch

### PR DESCRIPTION
**Pull request by CalvinRodo**
## To Test

Enter in a medical report. 
Navigate to /en/sessions to view the medical reports in the database 
Click on the Name of an Applicant to view their medical report

## Please note:
This is for the Trello Card: [Data is Saved to a Database](https://trello.com/c/Du6NkXXg/487-data-is-saved-to-a-database) and not the [MA View card](https://trello.com/c/UR2P2T1H/350-mas-can-view-what-applications-are-received-so-that-they-can-adjudicate-claims )
Although it does implement some of the styling of that card. For the sake of a smaller PR I've decided to do these in two seperate chunks

## Not Implemented
 - Search 
 - Proper Pagination
 - Empty sections are still displayed on MA View 
 - Filtering to display only submitted Reports

### CODE REVIEW

File: api/controllers/SessionsController.js [hunk 1]
```diff
@@ -5,52 +5,39 @@
...
-    let directory = path.resolve(__dirname, '../../sessions');
+  index: async function (req, res) {
+
+    const reports = await MedicalReport.findAll({limit: 10, attributes: ['id', 'firstName', 'lastName', 'socialInsuranceNumber']});
+    sails.log(`Query returned: ${JSON.stringify(reports)}`);
+    const viewModel = reports.map(x =>  {
+      return {
+        id: x.id,
+        dateRecieved: 'Not Implemented',
+        applicant: `${x.lastName}, ${x.firstName}`,
```
Comment (dsamojlenko) [inline api/controllers/SessionsController.js]: you can get this right from the model as `x.fullName`

File: api/controllers/SessionsController.js [hunk 2]
```diff
@@ -5,52 +5,39 @@
...
 const conditionHelper = require('../utils/ConditionHelpers');
 
 module.exports = {
-  index: function (req, res) {
-
-    let directory = path.resolve(__dirname, '../../sessions');
+  index: async function (req, res) {
+
+    const reports = await MedicalReport.findAll({limit: 10, attributes: ['id', 'firstName', 'lastName', 'socialInsuranceNumber']});
+    sails.log(`Query returned: ${JSON.stringify(reports)}`);
```
Comment (dsamojlenko) [inline api/controllers/SessionsController.js]: This is going to get noisy ... can we set the level to silly, ie: `sails.log.silly` so you can still see them if you really want, but we're not filling up stdout?

File: api/controllers/SessionsController.js [hunk 3]
```diff
@@ -5,52 +5,39 @@
...
-      })
-      .pipe(res);
-  },
+  view: async function (req, res) {
 
-  view: function (req, res) {
-    const filename = req.params.session;
-    const sessionFile = path.resolve(__dirname, '../../sessions/' + filename);
+    const medicalReport = await MedicalReport.findOne({ where: {id: req.params.session}});
+    sails.log(`medicalReport: ${JSON.stringify(medicalReport, 2)}`);
```
Comment (dsamojlenko) [inline api/controllers/SessionsController.js]: can call `.toJSON()` on the model

**Comment by jeffmaher** [comment]
Taking a look now.

File: api/controllers/SessionsController.js [hunk 4]
```diff
@@ -5,52 +5,39 @@
...
 
-    const medicalReport = JSON.parse(fs.readFileSync(sessionFile));
     const conditions = conditionHelper.getConditionsWithMedicationsAndTreatments(medicalReport);
```
Comment (dsamojlenko) [inline api/controllers/SessionsController.js]: This helper should be removed. Better to just build up the medical report with all dependencies in the main query, ie:

```
let medicalReport = await MedicalReport.findOne({
      where: {
        applicationCode: req.session.applicationCode
      },
      include: [
        {
          model: Condition,
          as: 'Conditions',
          where: { id: req.params.id },
          include: [
            {
              model: Document,
              as: 'Documents',
            },
            {
              model: Medication,
              as: 'Medications',
            },
            {
              model: Treatment,
              as: 'Treatments',
              include: [
                  {
                      model: Document,
                      as: 'Documents'
                  }
              ],
            },
          ]
        }
      ]
    });
```

Comment (dsamojlenko) [reply api/controllers/SessionsController.js]: ... You also shouldn't need the DocumentsHelper a few lines further down

File: config/routes/sessions.js [hunk 5]
```diff
@@ -13,17 +13,6 @@ module.exports.routes = {
...
-    controller: 'SessionsController',
-    action: 'download',
-    lang: 'en',
-    i18n: {
-      en: '/en/sessions/:session/download',
-      fr: '/fr/sessions/:session/download'
-    }
-  },
-
   'GET /en/sessions/:session/view': {
```
Comment (dsamojlenko) [inline config/routes/sessions.js]: TBD: we should probably change the route path (and controller name?)

File: views/pages/sessions.njk [hunk 6]
```diff
@@ -2,11 +2,42 @@
...
+        <input id="search" class="placeholder-gray-600 rounded-lg  pl-10 block w-full appearance-none leading-normal mb-10" type="text" placeholder="{{__("sessions.searchApplications")}}">
+        <div class="pointer-events-none absolute inset-y-0 left-0 pl-4 flex items-center">
+          <svg class="fill-current pointer-events-none text-gray-600 w-4 h-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M12.9 14.32a8 8 0 1 1 1.41-1.41l5.35 5.33-1.42 1.42-5.33-5.34zM8 14A6 6 0 1 0 8 2a6 6 0 0 0 0 12z"/></svg>
+        </div>
+      </div>
+    </div>
+  </form>
+  <table class="w-full">
+    <thead class="border-solid border-b-2 text-left">
+      <tr class="flex" >
```
Comment (dsamojlenko) [inline views/pages/sessions.njk]: shouldn't need to set flex on the tr? See the tailwind docs on fixed-width tables: https://tailwindcss.com/docs/table-layout/#fixed

File: views/pages/sessions.njk [hunk 7]
```diff
@@ -2,11 +2,42 @@
...
+        <th class="w-2/4">Applicant</th>
+        <th class="w-1/4">Last opened</th>
+      </tr>
+    </thead>
+    <tbody >
+      {% for report in reports %}
+        <tr class="flex p-2 {{ 'bg-gray-200' if loop.index % 2 == 0  }}">
+          <td class="w-1/4 ">{{ report.dateRecieved}}</td>
+          <td class="w-2/4 "><a href={{ route('sessions.view', { session: report.id})}}>{{ report.applicant}}</a> ({{report.sin }})</td>
+          <td class="w-1/4 ">{{ report.lastViewed}}</td>
```
Comment (dsamojlenko) [inline views/pages/sessions.njk]: inconsistent whitespace around variables (3 lines above)

(ie, `{{ report.lastViewed}}` should be `{{ report.lastViewed }}`

File: views/pages/sessions.njk [hunk 8]
```diff
@@ -2,11 +2,42 @@
...
-    {% endfor %}
-  </ul>
-
-{% endblock %}
\ No newline at end of file
+  <div class="my-4">
+    {% set pages = ((total/8)|round(0,'ceil')) %}
+    {% if total > 8 %}
+      {% for i in range(0, pages) %}
+        <a href="about/blank" class="m-1"><span class="sr-only">Navigate to page </span>{{ i + 1 }}<span class="sr-only"> of {{ pages }}</span></a>
```
Comment (dsamojlenko) [inline views/pages/sessions.njk]: I know `about/blank` is a placeholder, but maybe use `#`

**Comment by dsamojlenko** [comment]
Strange alignment inconsistencies:
image

File: views/pages/sessions.njk [hunk 9]
```diff
@@ -2,11 +2,42 @@
...
+    <thead class="border-solid border-b-2 text-left">
+      <tr class="flex" >
+        <th class="w-1/4">Date Recieved</th>
+        <th class="w-2/4">Applicant</th>
+        <th class="w-1/4">Last opened</th>
+      </tr>
+    </thead>
+    <tbody >
+      {% for report in reports %}
+        <tr class="flex p-2 {{ 'bg-gray-200' if loop.index % 2 == 0  }}">
```
Comment (dsamojlenko) [inline views/pages/sessions.njk]: can do the zebra-striping with pure tailwind using pseudo-classes, you just have to do it on the parent element: https://tailwindcss.com/docs/pseudo-class-variants/#odd-child

`<tbody class="even:bg-gray-200">`

Comment (CalvinRodo) [reply views/pages/sessions.njk]: can also be done in nunjucks https://mozilla.github.io/nunjucks/templating.html#cycler-item1-item2-itemn

Comment (CalvinRodo) [reply views/pages/sessions.njk]: Had to use the cycler as the pseudo-classes would not work.

File: api/controllers/SessionsController.js (hunk 1 above)
Comment (CalvinRodo) [reply api/controllers/SessionsController.js]: x.fullName is has the names in the wrong order

File: api/controllers/SessionsController.js (hunk 2 above)
Comment (CalvinRodo) [reply api/controllers/SessionsController.js]: done

File: views/pages/sessions.njk [hunk 10]
```diff
@@ -2,11 +2,43 @@
...
+        <div class="pointer-events-none absolute inset-y-0 left-0 pl-4 flex items-center">
+          <svg class="fill-current pointer-events-none text-gray-600 w-4 h-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M12.9 14.32a8 8 0 1 1 1.41-1.41l5.35 5.33-1.42 1.42-5.33-5.34zM8 14A6 6 0 1 0 8 2a6 6 0 0 0 0 12z"/></svg>
+        </div>
+      </div>
+    </div>
+  </form>
+  <table class="w-full">
+    <thead class="border-solid border-b-2 text-left">
+      <tr class="flex" >
+        <th class="w-1/4">Date Recieved</th>
```
Comment (dsamojlenko) [inline views/pages/sessions.njk]: `i` before `e` except after `c`. Received.

File: api/controllers/SessionsController.js (hunk 3 above)
Comment (CalvinRodo) [reply api/controllers/SessionsController.js]: toJSON doesn't return a string it returns a json object so you need to JSON.stringify the output of toJSON

File: config/routes/sessions.js (hunk 5 above)
Comment (CalvinRodo) [reply config/routes/sessions.js]: Agreed we should add it to the backlog

File: views/pages/sessions.njk (hunk 10 above)
Comment (CalvinRodo) [reply views/pages/sessions.njk]: fixed

File: views/pages/sessions.njk (hunk 8 above)
Comment (CalvinRodo) [reply views/pages/sessions.njk]: done

File: views/pages/sessions.njk (hunk 7 above)
Comment (CalvinRodo) [reply views/pages/sessions.njk]: done

File: views/pages/sessions.njk (hunk 6 above)
Comment (CalvinRodo) [reply views/pages/sessions.njk]: removed

**Comment by CalvinRodo** [comment]
> Strange alignment inconsistencies:
> image

I think that was a result of the padding, I modified it and can't replicate it anymore

File: api/controllers/SessionsController.js (hunk 4 above)
Comment (CalvinRodo) [reply api/controllers/SessionsController.js]: Done

**Comment by CalvinRodo** [comment]
Changes have been made ready for review.

**Pull request merged**